includes/validation/class-amp-invalid-url-post-type.php

```diff
@@ -585,92 +637,88 @@ public static function output_custom_column( $column_name, $post_id ) {
 				}
 				self::display_invalid_url_validation_error_counts_summary( $post_id );
 				break;
-			case AMP_Validation_Error_Taxonomy::REMOVED_ELEMENTS:
+			case AMP_Validation_Error_Taxonomy::FOUND_ELEMENTS_AND_ATTRIBUTES:
+				$items = array();
 				if ( ! empty( $error_summary[ AMP_Validation_Error_Taxonomy::REMOVED_ELEMENTS ] ) ) {
-					$items = array();
 					foreach ( $error_summary[ AMP_Validation_Error_Taxonomy::REMOVED_ELEMENTS ] as $name => $count ) {
 						if ( 1 === intval( $count ) ) {
 							$items[] = sprintf( '<code>%s</code>', esc_html( $name ) );
 						} else {
 							$items[] = sprintf( '<code>%s</code> (%d)', esc_html( $name ), $count );
 						}
 					}
-					echo implode( ', ', $items ); // WPCS: XSS OK.
-				} else {
-					esc_html_e( '--', 'amp' );
 				}
-				break;
-			case AMP_Validation_Error_Taxonomy::REMOVED_ATTRIBUTES:
 				if ( ! empty( $error_summary[ AMP_Validation_Error_Taxonomy::REMOVED_ATTRIBUTES ] ) ) {
-					$items = array();
 					foreach ( $error_summary[ AMP_Validation_Error_Taxonomy::REMOVED_ATTRIBUTES ] as $name => $count ) {
 						if ( 1 === intval( $count ) ) {
-							$items[] = sprintf( '<code>%s</code>', esc_html( $name ) );
+							$items[] = sprintf( '<code>[%s]</code>', esc_html( $name ) );
 						} else {
-							$items[] = sprintf( '<code>%s</code> (%d)', esc_html( $name ), $count );
+							$items[] = sprintf( '<code>[%s]</code> (%d)', esc_html( $name ), $count );
 						}
 					}
-					echo implode( ', ', $items ); // WPCS: XSS OK.
+				}
+				if ( ! empty( $items ) ) {
+					echo implode( ',<br/>', $items ); // WPCS: XSS OK.
 				} else {
 					esc_html_e( '--', 'amp' );
 				}
 				break;
 			case AMP_Validation_Error_Taxonomy::SOURCES_INVALID_OUTPUT:
 				if ( isset( $error_summary[ AMP_Validation_Error_Taxonomy::SOURCES_INVALID_OUTPUT ] ) ) {
-					$sources = array();
-					foreach ( $error_summary[ AMP_Validation_Error_Taxonomy::SOURCES_INVALID_OUTPUT ] as $type => $names ) {
-						foreach ( array_unique( $names ) as $name ) {
-							$sources[] = sprintf( '%s: <code>%s</code>', esc_html( $type ), esc_html( $name ) );
+					$sources = $error_summary[ AMP_Validation_Error_Taxonomy::SOURCES_INVALID_OUTPUT ];
+					$output  = array();
+
+					if ( isset( $sources['plugin'] ) ) {
+						$output[]     = '<details class="source">';
+						$plugin_names = array();
+						$plugin_slugs = array_unique( $sources['plugin'] );
+						foreach ( $plugin_slugs as $plugin_slug ) {
+							$plugin_data = get_plugin_data( WP_PLUGIN_DIR . '/' . $plugin_slug . '/' . $plugin_slug . '.php' );
```

Comment:
This won't work for plugins that don't have directories or which don't use the slug as the plugin's main PHP file.

![image](https://user-images.githubusercontent.com/134745/45593198-5c6f3f80-b935-11e8-8da6-3fb0caf2ed28.png)